openmdao/docs/feature_reference/building_blocks/components/file_wrap.rst

```diff
@@ -0,0 +1,820 @@
+
+Note also that we used the method ``reset_anchor`` to return the anchor to the
+beginning of the file before marking our new anchor. Subsequent calls to
+``mark_anchor`` start at the previous anchor and find the next instance of the
+anchor text. It is a good practice to reset your anchor unless you are looking for
+an instance of "B" that follows an instance of "A".
+
+You can also count backwards from the bottom of the file by passing a negative
+number. Here, the second instance of ``"INPUT"`` from the bottom brings you
+back to the first one.
+
+.. testcode:: Parse_Input
+
+    parser.reset_anchor()
+    parser.mark_anchor("INPUT", -2)
+    parser.transfer_var("99999", 1, 1)
+
+.. testcode:: Parse_Input
+    :hide:
+
+    from __future__ import print_function
+    for datum in parser.data:
+        print(datum)
+
+.. testoutput:: Parse_Input
+
+    INPUT
+    99999 7 3
+    INPUT
+    10.1 20.2 3.141592653589793
+    A B C
+
+There is also a method for replacing an entire array of values. Try
+replacing the set of three integers as follows:
+
+.. testcode:: Parse_Input
+
+    from numpy import array
+
+    array_val = array([123, 456, 789])
+
+    parser.reset_anchor()
+    parser.mark_anchor("INPUT")
+    parser.transfer_array(array_val, 1, 1, 3)
+
+.. testcode:: Parse_Input
+    :hide:
+
+    from __future__ import print_function
+    for datum in parser.data:
+        print(datum.rstrip())
+
+.. testoutput:: Parse_Input
+
+    INPUT
+    123 456 789
+    INPUT
+    10.1 20.2 3.141592653589793
+    A B C
+
+.. index:: transfer_array
+
+The method ``transfer_array`` takes four required inputs. The first is an array
+of values that will become the new values in the file. The second is the
+starting row after the anchor. The third is the starting field that will be
+replaced, and the fourth is the ending field. The new array replaces the
+block of fields spanned by the starting field and the ending field.
+
+You can also use the ``transfer_array`` method to `stretch` an existing
+array in a template to add more terms.
+
+.. testcode:: Parse_Input
+
+    from numpy import array
+
+    array_val = array([11, 22, 33, 44, 55, 66])
+
+    parser.reset_anchor()
+    parser.mark_anchor("INPUT")
+    parser.transfer_array(array_val, 1, 1, 3, sep=' ')
+
+.. testcode:: Parse_Input
+    :hide:
+
+    from __future__ import print_function
+    for datum in parser.data:
+        print(datum.rstrip())
+
+.. testoutput:: Parse_Input
+
+    INPUT
+    11 22 33 44 55 66
+    INPUT
+    10.1 20.2 3.141592653589793
+    A B C
+
+The named argument ``sep`` defines which separator to include between the
+additional terms of the array.
+
+The input file templating capability that comes with OpenMDAO is basic but quite
+functional. If you need a more powerful templating engine, particularly one that
+allows the inclusion of logic in your template files, then you may want to consider
+one of the community-developed engines, such as mako_ or django_.
+
```

Comment:
Wonder if we should even mention specific templating engines anymore, since we risk being hilariously out of date. I think Jinja2 is the big one now.

Reply:
ya.. maybe https://wiki.python.org/moin/Templating would be better